### PR TITLE
Implemented all yield tasks

### PR DESCRIPTION
**Pull request by juliaakor**
## Changes
- Added implementation of all functions

### Implemented Functions
1. get99BottlesOfBeer
2. getFibonacciSequence
3. depthTraversalTree
4. breadthTraversalTree
5. mergeSortedSequences

### CODE REVIEW

File: task/07-yield-tasks.js [hunk 1]
```diff
@@ -82,7 +104,16 @@ function* getFibonacciSequence() {
  *
  */
 function* depthTraversalTree(root) {
-    throw new Error('Not implemented');
+    const stack = [root]
+    let node
```
Comment (paperstick) [inline task/07-yield-tasks.js:108]: Is there any reason to create `node` variable there and not inside loop?

File: task/07-yield-tasks.js [hunk 2]
```diff
@@ -82,7 +104,16 @@ function* getFibonacciSequence() {
...
+
+    while(stack.length !== 0) {
```
Comment (paperstick) [inline task/07-yield-tasks.js:110]:
```suggestion
    while(stack.length) {
```
This check will do the same, but will look a bit cleaner

File: task/07-yield-tasks.js [hunk 3]
```diff
@@ -126,7 +164,24 @@ function* breadthTraversalTree(root) {
  *   [ 1, 3, 5, ... ], [ -1 ] => [ -1, 1, 3, 5, ...]
  */
 function* mergeSortedSequences(source1, source2) {
-    throw new Error('Not implemented');
+    const src1 = source1()
+    const src2 = source2()
+    let srcValue1 = src1.next().value
+    let srcValue2 = src2.next().value
+
+    while(srcValue1 !== undefined || srcValue2 !== undefined) {
```
Comment (paperstick) [inline task/07-yield-tasks.js:172]:
```suggestion
    while(!src1.done() && !src1.done()) {
```
This check will be a bit more understable

**Review by paperstick — CHANGES_REQUESTED** [review]

File: task/07-yield-tasks.js (hunk 1 above)
Comment (juliaakor) [reply task/07-yield-tasks.js:108]: To avoid creating a variable every iteration. Is it better to do this only with collections and not variables?

**Review by paperstick — APPROVED** [review]

**Pull request merged**